doc/source/whatsnew/v2.3.0.rst

```diff
@@ -35,6 +35,7 @@ Other enhancements
 - The semantics for the ``copy`` keyword in ``__array__`` methods (i.e. called
   when using ``np.array()`` or ``np.asarray()`` on pandas objects) has been
   updated to work correctly with NumPy >= 2 (:issue:`57739`)
+- :meth:`~Series.to_hdf` and :meth:`~DataFrame.to_hdf` now round-trip with ``StringDtype`` preserving the ``na_value`` but not necessarily the storage (:issue:`60663`)
```

Comment:
```suggestion
- :meth:`~Series.to_hdf` and :meth:`~DataFrame.to_hdf` now round-trip with ``StringDtype``  (:issue:`60663`)
```

I generally consider the storage of the string to be an implementation detail, not something the end user really needs to care that much about. So I'd prefer just to simplify further, but its not a blocker for me